### ISSUE TITLE
DatabaseIntrospection.get_table_list() shows objects outside of search_path

### ISSUE THREAD

**Issue by zenoran** [cockroachdb/django-cockroachdb#225]
[DatabaseIntrospection.get_table_list()](https://github.com/cockroachdb/django-cockroachdb/blob/426e9dc2ade8f75a69c3150a3e0dbf7b9bf6edd9/django_cockroachdb/introspection.py#L13-L16) needs to filter based off of name resolution logic in CockroachDB > 20.2 (where schemas were introduced).

Can produce an error specifically when you try to do a `manage.py migrate` and have the same tables in `public` (or another schema). 
 It tries to execute queries based off of name resolution and gets a failure if `search_path` is specified.  I believe this would cause other issues even if `search_path` isn't specified, since show tables would also show other schemas the user has access to and provide false information to Django.

Cleanest approach appears to be to filter based off of `current_schema()`:
```sql
SELECT table_name, type from [SHOW TABLES] where schema_name=current_schema();
```

**Comment by rafiss** [comment]
Thanks for the report. Which version of CockroachDB are you using?

**Comment by zenoran** [comment]
> Thanks for the report. Which version of CockroachDB are you using?

We are currently using 20.2.2

**Comment by rafiss** [comment]
Hmm the only relevant change I can think of is https://github.com/cockroachdb/cockroach/pull/59880 but that was a performance improvement. It's in 21.1 though, if you are able to try that let me know if it works. Oh also, which version of Django and django-cockroachdb are you using?

@timgraham do you know what other information from the django side would be helpful to debug this?

**Comment by timgraham** [comment]
The proposed change to the query that Django makes seems reasonable but I'm not intimately familiar with how CockroachDB is intended to work, so I'd say that's up to you to confirm.

PostgreSQL's [introspection query](https://github.com/django/django/blob/fde6fb28984a76d5bee794f4c0458eb25fe56fcd/django/db/backends/postgresql/introspection.py#L50-L56) has quite a bit more logic than [CockroachDB's](https://github.com/cockroachdb/django-cockroachdb/blob/426e9dc2ade8f75a69c3150a3e0dbf7b9bf6edd9/django_cockroachdb/introspection.py#L14) so it doesn't seem unreasonable that some changes to the latter might be needed.

**Comment by zenoran** [comment]
> Oh also, which version of Django and django-cockroachdb are you using?


We're still on Django 2.2 and I noticed a lot of the database connectivity was changed to support 3.x, so it's possible to see different behavior. Unfortunately, bigger corporation red tape makes it harder to bleeding edge with versions but it appears that would be an issue with the way it was written for 3.x as well within that version of cockroach.

That being said, we ran into too many other inconsistencies with the way schemas were working within 20.2 so we decided to back off and use databases as separation instead until we can get to 21.x. The way the documentation was written though, it sounded like schemas were the recommended direction for future versions so it's probably in the best interest of this plugin to stay compatible with that approach, so I wanted to at least raise awareness to the issue.

**Comment by rafiss** [comment]
Oh sorry, I missed the part of your original issue report where you linked to the `get_table_list` function. 

Though actually I'm a bit confused. I thought django-cockroachdb already _was_ using the same [introspection query](https://github.com/django/django/blob/fde6fb28984a76d5bee794f4c0458eb25fe56fcd/django/db/backends/postgresql/introspection.py#L50-L56) that the postgres adapter uses (based on https://github.com/cockroachdb/cockroach/issues/57924). This is also why I made my earlier comment pointing to https://github.com/cockroachdb/cockroach/pull/59880 -- I thought this issue was related to how this Postgres query works.

One thing I'd point out is that the postgres adapter filters by `pg_table_is_visible`, which includes all tables in the current `search_path`. So technically, I believe it's possible for multiple tables with the same name to be returned even if you're not using CockroachDB, so this might be an issue that needs to be fixed in Django itself.

Anyway, as long as you don't care about the whole `search_path`, then yes, the change to filter by `current_schema()` that you propose sounds reasonable. Want to make a PR?

**Comment by timgraham** [comment]
@zenoran, could you confirm if the issue still exists if you remove `get_table_list()` as in https://github.com/cockroachdb/django-cockroachdb/pull/227? If so, could you give queries to create this scenario in the database?

**Comment by timgraham** [comment]
@zenoran, 931914a427859bfaca7e0dc0de8d996e47a016f3 is released in django-cockroachdb 3.2.2. If this is still an issue, could you please reopen with steps to reproduce?
